knowrob_srdl/prolog/knowrob/urdf_parser.pl

```diff
@@ -0,0 +1,357 @@
+/*
+  Copyright (C) 2018 Georg Bartels
+  All rights reserved.
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions are met:
+      * Redistributions of source code must retain the above copyright
+        notice, this list of conditions and the following disclaimer.
+      * Redistributions in binary form must reproduce the above copyright
+        notice, this list of conditions and the following disclaimer in the
+        documentation and/or other materials provided with the distribution.
+      * Neither the name of the <organization> nor the
+        names of its contributors may be used to endorse or promote products
+        derived from this software without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  DISCLAIMED. IN NO EVENT SHALL <COPYRIGHT HOLDER> BE LIABLE FOR ANY
+  DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+:- module(urdf_parser,
+    [
+      load_urdf_file/1,
+      load_urdf_param/1,
+      load_urdf_string/1,
+      robot_name/1,
+      root_link_name/1,
+      link_names/1,
+      link_parent_joint/2,
+      link_child_joints/2,
+      link_inertial_origin/2,
+      link_inertial_mass/2,
+      link_inertial_inertia/2,
+      link_num_visuals/2,
+      link_visual_type/3,
+      link_visual_origin/3,
+      link_visual_geometry/3,
+      link_material_name/3,
+      link_material_color/3,
+      link_material_texture/3,
+      link_num_collisions/2,
+      link_collision_type/3,
+      link_collision_name/3,
+      link_collision_origin/3,
+      link_collision_geometry/3,
+      joint_names/1,
+      joint_type/2,
+      joint_child_link/2,
+      joint_parent_link/2,
+      joint_axis/2,
+      joint_origin/2,
+      joint_lower_pos_limit/2,
+      joint_upper_pos_limit/2,
+      joint_velocity_limit/2,
+      joint_effort_limit/2,
+      joint_calibration_rising/2,
+      joint_calibration_falling/2,
+      joint_dynamics_damping/2,
+      joint_dynamics_friction/2,
+      joint_mimic_joint_name/2,
+      joint_mimic_multiplier/2,
+      joint_mimic_offset/2,
+      joint_safety_lower_limit/2,
+      joint_safety_upper_limit/2,
+      joint_safety_kp/2,
+      joint_safety_kv/2
+    ]).
```

Comment:
as pointed out earlier, some of the rules could be joined. But as long as these predicates are only used internally, I don't mind much